torch/testing/_internal/schema_check_mode.py

```diff
@@ -72,68 +80,83 @@ def unwrap(e):
 
         # Process arguments with outputs
         for arg in func._schema.arguments:
             name = standardize_name(arg.name)
             if arguments.get(name) is not None:
-                arg_alias_pairs = arg_alias_pairs_map[arg.name]
-                before = tree_flatten(cloned_arguments.get(name))[0]
-                after = tree_flatten(arguments.get(name))[0]
-                u_values = tree_map(unwrap, after)
-                u_out = tree_map(unwrap, out)
-                u_out = u_out if isinstance(u_out, tuple) else (u_out, )
-                if any([has_mutated(i, j) for i, j in zip(before, after)]):
+                arg_alias_pairs = arg_alias_pairs_map[name]
+                before = cloned_arguments.get(name)
+                md = cloned_metadata.get(name)
+                after = arguments.get(name)
+                for v in after:
+                    for i in range(len(u_out)):
+                        for j in range(len(u_out[i])):
+                            if has_aliased(v, u_out[i][j]):
+                                if not is_aliasing(func._schema.returns[i].alias_info, arg_alias_pairs):
+                                    raise RuntimeError(f'Argument {name} is not defined to alias output but was aliasing')
+                                else:
+                                    self.aliasing.append(Aliasing(func._schema.name, name, f"output_{i}"))
+                                    out_alias_pairs_map[i].add(name)
+                if any([has_mutated(i, j, k) for i, j, k in zip(before, after, md)]):
                     if not is_mutable(arg_alias_pairs):
                         raise RuntimeError(f"Argument {name} is not defined as mutable but was mutated")
                     else:
                         self.mutated.append(Mutation(func._schema.name, name))
-                for v in u_values:
-                    for j in range(len(u_out)):
-                        if has_aliased(v, u_out[j]):
-                            if not is_aliasing(func._schema.returns[j].alias_info, arg_alias_pairs):
-                                raise RuntimeError(f'Argument {name} is not defined to alias output but was aliasing')
-                            else:
-                                self.aliasing.append(Aliasing(func._schema.name, name, f"output_{j}"))
-                                out_alias_pairs_map[j].add(name)
 
         # Aliasing between outputs
         for i, j in combinations(range(len(func._schema.returns)), 2):
-            i_values = tree_map(
-                unwrap,
-                tree_flatten(out[i])[0])
-            j_values = tree_map(
-                unwrap,
-                tree_flatten(out[j])[0])
-            if are_args_aliasing(i_values, j_values) and not bool(len(out_alias_pairs_map[i] & out_alias_pairs_map[j])):
+            if are_args_aliasing(u_out[i], u_out[j]) and not bool(len(out_alias_pairs_map[i] & out_alias_pairs_map[j])):
```

Comment:
Separate this if statement out into multiple lines, and give 
`bool(len(out_alias_pairs_map[i] & out_alias_pairs_map[j]))` a variable name